### ISSUE TITLE
gcc8: Unaligned structure 

### ISSUE THREAD

**Issue by nmorey** [openucx/ucx#2400]
When trying to compile v1.2.2 or 1.3.0-rc3 using gcc 8.0.1, compilation fails with this error:

[   25s] In file included from /home/abuild/rpmbuild/BUILD/ucx-1.3.0-rc3/src/uct/ib/ud/base/ud_iface.h:22,
[   25s]                  from ib/ud/verbs/ud_verbs.c:19:
[   25s] /home/abuild/rpmbuild/BUILD/ucx-1.3.0-rc3/src/uct/ib/ud/base/ud_def.h:177:1: error: alignment 1 of 'struct uct_ud_send_skb_inl' is less than 64 [-Werror=packed-not-aligned]
[   25s]  } UCS_S_PACKED uct_ud_send_skb_inl_t;
[   25s]  ^
[   25s] cc1: all warnings being treated as errors
[   25s] make[2]: *** [Makefile:1601: ib/ud/verbs/libuct_la-ud_verbs.lo] Error 1

**Comment by yosefe** [comment]
duplicated by #2401 (which contains more details how to reproduce)

**Comment by hoopoepg** [comment]
this is different issue & we can't reproduce it in out environment (using Fedora 28 in docker)

could we request more details about failed build config? gcc version (including build number) and OS version

thank you

**Comment by hoopoepg** [comment]
@nmorey could you describe your environment?
and command line to reproduce issue (may be used any specific arguments/flags?)

thank you

**Comment by nmorey** [comment]
@hoopoepg Trying to build on SUSE Tumbleweed with gcc 8
I think TW still uses gcc 7 by default but the SUSE guys are notifying package failing with the next gcc upgrade.
```
abuild@portia:~> gcc -v
Using built-in specs.
Reading specs from /usr/lib64/gcc/x86_64-suse-linux/8/defaults.spec
COLLECT_GCC=/usr/bin/gcc
COLLECT_LTO_WRAPPER=/usr/lib64/gcc/x86_64-suse-linux/8/lto-wrapper
OFFLOAD_TARGET_NAMES=hsa:nvptx-none
Target: x86_64-suse-linux
Configured with: ../configure --prefix=/usr --infodir=/usr/share/info --mandir=/usr/share/man --libdir=/usr/lib64 --libexecdir=/usr/lib64 --enable-languages=c,c++,objc,fortran,obj-c++,ada,go --enable-offload-targets=hsa,nvptx-none=/usr/nvptx-none, --without-cuda-driver --disable-werror --with-gxx-include-dir=/usr/include/c++/8 --enable-ssp --disable-libssp --disable-libvtv --disable-cet --disable-libcc1 --enable-plugin --with-bugurl=http://bugs.opensuse.org/ --with-pkgversion='SUSE Linux' --with-slibdir=/lib64 --with-system-zlib --enable-__cxa_atexit --enable-libstdcxx-allocator=new --disable-libstdcxx-pch --enable-version-specific-runtime-libs --with-gcc-major-version-only --enable-linker-build-id --enable-linux-futex --enable-gnu-indirect-function --program-suffix=-8 --without-system-libunwind --enable-multilib --with-arch-32=x86-64 --with-tune=generic --build=x86_64-suse-linux --host=x86_64-suse-linux
Thread model: posix
gcc version 8.0.1 20180312 (experimental) [trunk revision 258445] (SUSE Linux) 
```

I'm building a RPM with these settings
```
%configure --disable-static --without-avx --docdir="%_docdir/%name"
make %{?_smp_mflags} V=1
```

**Comment by amaslenn** [comment]
@hoopoepg to reproduce this issue need to install `libibverbs-devel` package (UCX depends on it).
Log from Fedora env: https://kojipkgs.fedoraproject.org//work/tasks/1626/25781626/build.log

**Comment by shamisp** [comment]
The issues was resolved in master and v1.3. Thank you for bringing this up.
